internal/authz/oidc_test.go

```diff
@@ -940,6 +1012,793 @@ func TestOIDCProcess(t *testing.T) {
 	}
 }
 
+func TestPostClientAuthenticationMethod(t *testing.T) {
+
+	testOIDCProcessReusable(t, postOIDCConfig)
+}
+
+func TestEmptyClientAuthenticationMethod(t *testing.T) {
+
+	testOIDCProcessReusable(t, emptyAuthMethodOIDCConfig)
+}
+
+func TestJWTClientAuthenticationMethodUnauthenticated(t *testing.T) {
+
+	jwkPriv, _ := newKeyPair(t)
+
+	clock := oidc.Clock{}
+	sessions := &mockSessionStoreFactory{store: oidc.NewMemoryStore(&clock, time.Hour, time.Hour)}
+	store := sessions.Get(jwtAuthMethodOIDCConfig)
+	tlsPool := internal.NewTLSConfigPool(context.Background())
+	h, err := NewOIDCHandler(jwtAuthMethodOIDCConfig, tlsPool,
+		oidc.NewJWKSProvider(newConfigFor(basicOIDCConfig), tlsPool), sessions, clock,
+		oidc.NewStaticGenerator(newSessionID, newNonce, newState, newCodeVerifier))
+	require.NoError(t, err)
+
+	idpServer := newServer(wellKnownURIs)
+	h.(*oidcHandler).httpClient = idpServer.newHTTPClient()
+
+	ctx := context.Background()
+
+	idpServer.Start()
+	t.Cleanup(func() {
+		idpServer.Stop()
+		require.NoError(t, store.RemoveSession(ctx, sessionID))
+	})
+
+	idpServer.tokensResponse = &idpTokensResponse{
+		IDToken:     newJWT(t, jwkPriv, jwt.NewBuilder().Audience([]string{"test-client-id"}).Claim("nonce", newNonce)),
+		AccessToken: "access-token",
+		TokenType:   "Bearer",
+	}
+	idpServer.statusCode = http.StatusOK
+
+	require.NoError(t, store.SetAuthorizationState(ctx, sessionID, validAuthState))
+
+	t.Run("callback request ", func(t *testing.T) {
+		resp := &envoy.CheckResponse{}
+		require.NoError(t, h.Process(ctx, callbackRequest, resp))
+		require.Equal(t, int32(codes.Unauthenticated), resp.GetStatus().GetCode())
+		requireStandardResponseHeaders(t, resp)
+		requireStoredTokens(t, store, sessionID, false)
+	})
+}
+
+func testOIDCProcessReusable(t *testing.T, oidcConfig *oidcv1.OIDCConfig) {
```

Comment:
What is the main difference between this method and the existing `TestOIDCProcess` one? Is it that this one just gets the OIDC config as an argument?
Can we refactor the tests so that we reuse the code and add these cases to a single one, instead of duplicating the test logic?